### PR TITLE
[WIP] content-type header parsing for server files

### PR DESCRIPTION
**Pull request by gdj0nes**
### Motivation and context
Use the `Content-Header` to assist in determining the file type of a remote file when the url is missing an extension

### How has this been tested?
TODO

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat/apps/engine/task.py [hunk 1]
```diff
@@ -178,25 +189,40 @@ def _validate_data(counter):
...
+                    if content_type is not None:
+                        extension = get_file_extentsion_from_content_type(content_type)
+                        slogger.glob.info("Adding extension {} based on Content-Teader".format(extension))
+                        name += extension
+
+                if name in local_files:
+                    raise FileExistsError("filename collision: {}".format(name))
+
+                with open(os.path.join(upload_dir, name), 'wb') as tfp:
+                    while True:
```
Comment (gdj0nes) [inline cvat/apps/engine/task.py:220]: Actual code changes are in here

File: cvat/apps/engine/media_extractors.py [hunk 2]
```diff
@@ -20,6 +20,9 @@
 
 from cvat.apps.engine.mime_types import mimetypes
 
+def get_file_extentsion_from_content_type(content_type: str):
```
Comment (gdj0nes) [inline cvat/apps/engine/media_extractors.py:23]: Sometime this adds less conventional extensions like .jpe

File: cvat/apps/engine/task.py [hunk 3]
```diff
@@ -178,25 +189,40 @@ def _validate_data(counter):
...
+                        block = fp.read(8192)
+                        if not block:
+                            break
+                        tfp.write(block)
```
Comment (gdj0nes) [inline cvat/apps/engine/task.py:224]: Adding file extension based on content header happens here

**Comment by gdj0nes** [comment]
In a slightly unrelated not has their been consideration of using a standardized code like black. I think it would help make diffs more interpretable

**Comment by nmanovic** [comment]
@gdj0nes , thanks for the contribution. What is the status of the PR? When will it be ready?

**Comment by nmanovic** [comment]
@Marishka17 , do you think we can improve the PR and merge? Please give your opinion about the implemented functionality.

**Comment by nmanovic** [comment]
I will close the PR. @Marishka17 prepared another PR based on this one.

**Pull request closed without merging**